ground/src/main/java/com/google/android/ground/MainViewModel.kt

```diff
@@ -74,7 +80,8 @@ constructor(
     return signInState.result.fold(
       {
         when (signInState.state) {
-          SignInState.State.SIGNED_IN -> onUserSignedIn(it!!)
+          SignInState.State.SIGNED_IN ->
+            rxObservable(defaultDispatcher) { send(onUserSignedIn(it!!)) }
```

Comment:
What is the purpose of the added constructs? How do we know when to use them?

Reply:
The default ktdoc is sufficient. I was mostly concerned about how do we ensure that we apply these changes as we migrate more code to coroutines.

Reply:
Makes sense.. Tests won't work correctly without setting the dispatcher, and nothing will be emitted without `send`, so correctness elsewhere can be enforced by unit and manual testing. Lmk if you have another idea!